src/app/overlay.service.ts

```diff
@@ -1,34 +1,121 @@
-import { GlobalPositionStrategy, Overlay, OverlayConfig } from '@angular/cdk/overlay';
-import { ComponentPortal } from '@angular/cdk/portal';
-import { Injectable } from '@angular/core';
+import {
+  GlobalPositionStrategy,
+  Overlay,
+  OverlayConfig
+} from '@angular/cdk/overlay';
+import { ComponentPortal, PortalInjector } from '@angular/cdk/portal';
+import { Injectable, ComponentRef, Injector } from '@angular/core';
 import { OverlayComponent } from './overlay/overlay.component';
+import { CdkOverlayOrigin } from '@angular/cdk/overlay';
+import { PositionStrategy } from '@angular/cdk/overlay';
+import { ConnectionPositionPair } from '@angular/cdk/overlay';
+import { OverlayRef } from '@angular/cdk/overlay';
+import { AppOverlayRef } from './overlay.ref';
 
+interface AppOverlayConfig {
+  panelClass?: string;
+  hasBackdrop?: boolean;
+  backdropClass?: string;
+  height?: number;
+  width?: number;
+  minHeight?: number;
+  maxHeight?: number;
+  positionStrategy?: PositionStrategy;
+}
 
 @Injectable({
   providedIn: 'root'
 })
 export class OverlayService {
+  constructor(private overlay: Overlay, private injector: Injector) {}
 
-  constructor(private overlay: Overlay) { }
-
-  open() {
+  open(position: CdkOverlayOrigin) {
+    const overlayConfig: AppOverlayConfig = {
+      hasBackdrop: false,
```

Comment:
```suggestion
      hasBackdrop: true,
```

Reply:
Sorry about that!

By the way, the sass for changing the backdrop color would be this:

`  
.cdk-overlay-backdrop.cdk-overlay-backdrop-showing {
    &.dark-backdrop {
      background-color: rgb(39, 43, 48);
      opacity: 0.5;
    }
  }
`